DiscordPy/bot.py

```diff
@@ -70,11 +72,12 @@ async def getHelp(ctx, *arg):
 
 @bot.command(name='meme')
 async def getMeme(ctx, *arg):
-    if (len(imageList) == 0):
-        hot_python = subreddit.new(limit=20)
+    delta_time = datetime.datetime.now() - prev_time
+    delta_seconds = delta_time.total_seconds()
+    if (len(imageList) == 0 or delta_seconds >= 60):
+        hot_python = subreddit.new(limit=100)
         for i in hot_python:
-            extension = i.url[-3:]
-            if (extension == "jpg"):
+            if i.link_flair_text in {"Meme", "Loss", "Shitpost"} and i.url[-3:] in {"jpg", "png"}:
                 imageList.append(i.url)
     val = random.randint(0, len(imageList) - 1)
     response = imageList[val]

```

Comment:
Could we potentially pop a random element from the list, rather than removing everything and re-appending?
Other than that, looks great!

Reply:
That's a great idea, pop the element when it gets called. I will need to only add new meme images only if the time that it was created is past when the 'prev_call_meme_time' was last called